cmd/node-agent/README.md

```diff
@@ -0,0 +1,87 @@
+# Node Agent
+
+## Description
+
+The Node Agent is a process that runs on each node of a Kubernetes cluster. The
+purpose of this process is to monitor a specific annotation on the node object.
+This annotation holds a JSON object which describes the version and configuration
+to be used for the on-host kubelet.
+
+This is useful for upgrading your on-host kubelet without ever having to ssh
```

Comment:
nit: In the design doc we've changed "on-host kubelet" to "system-hosted kubelet". Maybe change here for consistency.